src/base/random.hh

```diff
@@ -62,12 +62,20 @@ class Random : public Serializable
 {
 
   public:
-
     /**
      * @ingroup api_base_utils
      */
     std::mt19937_64 gen;
 
+    static inline uint64_t globalSeed = 5489;
```

Comment:
Nit: you don't need the inline kw

Reply:
With gcc 10.2.1 it complains if I do not have it. : `ISO C++ forbids in-class initialization of non-const static member ‘gem5::Random::globalSeed’`

Reply:
We don't support gcc 10: https://github.com/gem5/gem5/blob/stable/RELEASE-NOTES.md#compiler-and-os-support

Reply:
> With gcc 10.2.1 it complains if I do not have it. : ISO C++ forbids in-class initialization of non-const static member ‘gem5::Random::globalSeed’

Even if we don't support gcc10, I wouldn't disregard this. And it makes sense I forgot this only works for const variables. As inline is simply saying the compiler to not enforce the ODR, my fear is every translation units ends up with a different version of globalSeed. So the solution is to define/initialize in source file (random.cc). I think that is the safest option that makes everyone happy

Reply:
I can go back to the old way but I thought C++17 introduced static inline variables notably for this purpose.

From https://en.cppreference.com/w/cpp/language/inline, it suggests that since C++17, an inline variable without external linkage (i.e., `static inline`) may not have several definitions (in different translation units), because it may have several definitions only if it has external linkage.

I'll roll back to keep it safe though.

Reply:
Also, not to nitpick but the link to the release notes shows `As of this release gem5 support Clang version 6 to 16 and GCC version **10** to 13` for gem5 24.0.0. Maybe the plan is to drop it for the next version though (?).

Reply:
Oh, thanks for pointing that out! That was my confusion 🙂. I get all sorts of confused between clang and gcc versions anymore...

Reply:
I'm answering this comment because actually tried the old style and it fails some tests, like :

`/fast-scratch/peraisa/gem5_fix/build/VEGA_X86/gem5.debug -d /tmp/gem5outrefaxyp2 -re --silent-redirect /fast-scratch/peraisa/gem5_fix/configs/example/apu_se.py --download-resource square-gpu-test --download-dir /fast-scratch/peraisa/gem5_fix/tests/gem5/resources --reg-alloc-policy=dynamic -n3 -c /fast-scratch/peraisa/gem5_fix/tests/gem5/resources/square-gpu-test`

I'm debugging it but the problem is the instance of `Random` in an anonymous namespace in `syscall_emul.hh` (`se_prng`). The static initialization of the libc initaliazes multiples objects through static initialization (which is already strange), and deletes one object two times, so the assert where it should be in the `instances` vector if it is deleted fires.

One way to make it work is to make `instances` a `static inline` member, but it's not obvious why changing `instances` is going to fix the behavior of multiples instances of `se_prng` being instantiated, unless there are also multiple versions of `instances` being created. I have seen this kind of mixups in the past where several dynamic libraries have their own version of a static variable (`static inline` is also supposed to solve that...I think...At least we used it for that purpose). I'll update once I get a clearer picture.

Reply:
`static inline` is fine with me. You may want to include a pointer to this PR in a comment for the future. Or, we can rely on git blame :)